keras/utils/conv_utils.py

```diff
@@ -49,6 +49,31 @@ def convert_data_format(data_format, ndim):
         'Expected values are ["channels_first", "channels_last"]')
 
 
+def normalize_positive_tuple(value, n, name):
+  """Transforms a single positive integer or iterable of positive integers into an
+    integer tuple.
+
+  Args:
+    value: The value to validate and convert. Could a positive int, or any iterable
+      of positive ints.
+    n: The size of the tuple to be returned.
+    name: The name of the argument being validated, e.g. "pool_size" or "kernel_size".
+    This is only used for format error messages.
+
+  Returns:
+    A tuple of n positive integers.
+
+  Raises:
+    ValueError: If something else than an postive int/long or iterable thereof was passed.
+  """
+  value_tuple = normalize_tuple(value, n, name)
+  for value in value_tuple:
+    if value <= 0:
+      raise ValueError(f'The `{name}` argument must be a positive int or a tuple of positive integers. '
+                       f'Received: {value}.')
+  return value_tuple
+
+
 def normalize_tuple(value, n, name):
```

Comment:
Is there any usage of negative number for this method (eg -1 for unknown shape). If all the intended use case for this method is for positive int, then let's just update this method instead.

Reply:
For `stride` and `padding`, it may contain zero values, such as (2, 1, 0). But, I am not sure for the cases of `size` and `dilation_rate`.

Reply:
kernel_size should always be positive. dilation_rate is always positive too. For stride, I think it should be positive and non-zero as well (0 mean the window doesn't even move?). Padding is not validated by this function (checked by normalize_padding() instead).

With that, I think let's just update the existing normalize_tuple with an positive number check instead.

Reply:
Instead of inserting a positive number check into normalize_tuple, I add a keyword argument to the function. And the caller can check the normalized tuple according to their needs.